src/commands.ts

```diff
@@ -292,70 +295,54 @@ export const createBlock = (
   }
 }
 
-export const undoFootnoteDelete = (
-  tr: Transaction,
-  footnote: NodeWithPos,
-  position: number
+export const insertInlineTableFootnote = (
+  state: ManuscriptEditorState,
+  dispatch?: Dispatch
 ) => {
-  const updatedAttrs = {
-    ...footnote.node.attrs,
-    dataTracked: null,
+  const $pos = state.selection.$to
+  const table = findParentNodeOfTypeClosestToPos($pos, schema.nodes.table)
+  if (!table) {
+    return false
+  }
+  if (!dispatch) {
+    return true
   }
-  tr.setNodeMarkup(position, undefined, updatedAttrs, footnote.node.marks)
+  return insertInlineFootnote(state, dispatch)
 }
 
-export const insertGeneralFootnote = (
-  tableElementNode: ManuscriptNode,
-  position: number,
-  view: ManuscriptEditorView,
-  tableElementFooter?: NodeWithPos[]
+export const insertGeneralTableFootnote = (
+  state: ManuscriptEditorState,
+  dispatch?: Dispatch
 ) => {
-  const { state, dispatch } = view
-  const paragraph = state.schema.nodes.paragraph.create({
-    placeholder: 'Add general note here',
-  })
-  const generalNote = state.schema.nodes.general_table_footnote.create({}, [
-    paragraph,
-  ])
-  const tableColGroup = findChildrenByType(
-    tableElementNode,
-    schema.nodes.table_colgroup
-  )[0]
-  const table = findChildrenByType(tableElementNode, schema.nodes.table)[0]
-  const tr = state.tr
-  const pos = tableElementFooter?.length
-    ? position + tableElementFooter[0].pos + 2
-    : position +
-      (!tableColGroup
-        ? table.pos + table.node.nodeSize
-        : tableColGroup.pos + tableColGroup.node.nodeSize)
-
-  if (tableElementFooter?.length) {
-    if (isDeleted(tableElementFooter[0].node)) {
-      const tableElementFooterPos = tr.mapping.map(
-        position + tableElementFooter[0].pos + 1
-      )
-
-      //Restore the deleted/ rejected tablefooter by clearing the 'dataTracked' attribute (setting it to null)
-      undoFootnoteDelete(tr, tableElementFooter[0], tableElementFooterPos)
-    }
-
-    tr.insert(pos, generalNote as ManuscriptNode)
-  } else {
-    const tableElementFooter = schema.nodes.table_element_footer.create(
-      {
-        id: generateNodeID(schema.nodes.table_element_footer),
-      },
-      [generalNote]
-    )
-    tr.insert(pos, tableElementFooter)
+  const $pos = state.selection.$to
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const container = findParentNodeOfTypeClosestToPos(
+    $pos,
+    schema.nodes.table_element
+  )!
+
+  const existing = findChildrenByType(
+    container.node,
+    schema.nodes.general_table_footnote
+  )
+  if (existing) {
+    return false
```

Comment:
So you want to block adding a general note even if there is a pending deleted one?

Reply:
I think that's reasonable, you can always reject the deletion.

Reply:
Yes, that seems reasonable. This will also address LEAN-4020.